deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,239 @@
+"""The class performance imbalance check."""
+import typing as t
+from functools import partial
+from collections import defaultdict
+
+import pandas as pd
+import numpy as np
+from sklearn.metrics import precision_score, recall_score, get_scorer
+from sklearn.utils.multiclass import unique_labels as get_unique_labels
+
+from deepchecks import SingleDatasetBaseCheck, Dataset, CheckResult, ConditionResult
+from deepchecks.metric_utils import task_type_validation, ModelType
+from deepchecks.string_utils import format_percent
+from deepchecks.utils import DeepchecksValueError
+
+
+__all__ = ['ClassPerformanceImbalanceCheck']
+
+
+MetricFunc = t.Callable[
+    [object, pd.DataFrame, pd.Series], # model, features, labels
+    t.Dict[t.Hashable, t.Union[float, int]] # score for each label
```

Comment:
doesn't metric function returns a single number?

Reply:
1. some scorers will return dict of values per label if to pass `average=None`
2. do not we want to calc metric per label (class)?

Reply:
yes, but if the user passes custom scorer I guess it will not return a dict.
@noamzbr if we filter only specific labels and run metrics on them it will work? or it will need also the false positive for this class?

Reply:
yup, if the user passes a scorer function without applying `avarage=None`, then check will fail within the `_execute_alternative_metrics` method, with the next message error:
```
Check {check_name} expecting that alternative metrics will return
not empty instance of 'Mapping[Hashable, float|int]', but got {type(value)}
```

what I mean is that it is the user's responsibility to follow the check API contract.

Reply:
I'd edit it to something more readable, `metrics should return a dictionary of score per label class`

Reply:
it's not metrics it's scorers (difference is scorers receive (model, features, labels) and not (pred, labels))

Reply:
You're right, we should rename it globally and make clear we're using scorers rather than metrics.

Unless you or @matanper think the standard of using scorers is wrong

Reply:
I have no opinion in this matter, so let's go for scores

Reply:
renamed, but only in this check. For others I think it is better to create separate PR (also created an issue for this - [MLC-301](https://deepchecks.atlassian.net/browse/MLC-301))